libs/langchain/langchain/smith/evaluation/runner_utils.py

```diff
@@ -1330,6 +1335,24 @@ def run_on_dataset(
     return container.finish(batch_results, verbose=verbose)
 
 
+def _get_default_revision_id() -> Optional[str]:
```

Comment:
Could you add a description of what `LANGCHAIN_REVISION_ID` is? And which `git` repo is this for?

Reply:
Do we need to memoize the result if `git describe --tags --dirty` fails? e.g., no git installed or no .git found?

Reply:
Ya - he's implementing in the SDK also! we memoize there